examples/dataset/crossing_pedestrian_detection/crossing_pedestrian_detection/upload_results.py

```diff
@@ -0,0 +1,307 @@
+                if inf.score is not None:
+                    previous_conf = inf.score
+            else:
+                if inf.score is not None:
+                    previous_conf = inf.score
+        fail_inf = inf.failed_to_infer if not in_observation_state and previous_conf < 0 else False
+        processed_inferences.append(
+            ScoredPedestrianBoundingBox(
+                ped_id=inf.ped_id,
+                top_left=inf.top_left,
+                bottom_right=inf.bottom_right,
+                frame_id=inf.frame_id,
+                occlusion=inf.occlusion,
+                time_to_event=inf.time_to_event,
+                failed_to_infer=fail_inf,
+                score=confidence_score,
+                observation=in_observation_state,
+                label=inf.label,
+            ),
+        )
+
+    return processed_inferences
+
+
+def process_inf_data(action_model_name: str, detection_model_name: str) -> Dict[str, List[ScoredPedestrianBoundingBox]]:
+    model_pkl_name = f"s3://{BUCKET}/{DATASET}/raw/jaad_{action_model_name}_{detection_model_name}_database.pkl"
+    with smart_open(model_pkl_name, "rb") as inf_file:
+        inf_annotations = pickle.load(inf_file)
+
+    results_pkl = f"s3://{BUCKET}/{DATASET}/results/raw/{action_model_name}_{detection_model_name}.pkl"
+    with smart_open(results_pkl, "rb") as results_file:
+        results = pickle.load(results_file)
+        predictions: Dict[str, Dict[int, Dict[str, float]]] = defaultdict(lambda: defaultdict(dict))
+        for pid, tte, y, img in zip(results["pid"], results["tte"], results["y"], results["image"]):
+            pid = pid[0][0]
+            frame_id = int(Path(img[0]).stem)
+            predictions[pid][frame_id]["tte"] = tte[0]
+            predictions[pid][frame_id]["confidence"] = y[0]
+
+    scored_bboxes_per_ped: Dict[str, List[ScoredPedestrianBoundingBox]] = {}
+    for filename in inf_annotations:
+        bboxes_per_file = process_ped_annotations(inf_annotations[filename]["ped_annotations"])
+        scored_bboxes_lst = []
+        for ped_id, bboxes in bboxes_per_file.items():
+            for bbox in bboxes:
+                if bbox.ped_id not in predictions or bbox.frame_id not in predictions[ped_id]:
+                    failed_to_infer = True
+                else:
+                    failed_to_infer = False
+                    confidence = predictions[bbox.ped_id][bbox.frame_id]["confidence"]
+                    time_to_event = predictions[bbox.ped_id][bbox.frame_id]["tte"]
+
+                scored_bboxes_lst.append(
+                    ScoredPedestrianBoundingBox(
+                        ped_id=bbox.ped_id,
+                        top_left=bbox.top_left,
+                        bottom_right=bbox.bottom_right,
+                        frame_id=bbox.frame_id,
+                        occlusion=bbox.occlusion,
+                        time_to_event=time_to_event if not failed_to_infer else None,
```

Comment:
based on [this example](https://trunk.kolena.io/nitesh-dev/dataset?datasetId=67), a lot of the FN, TN bbox has no time_to_event or null value, is this expected?

Reply:
Yes, the FNs and TNs are ground truth bboxes which do not have these fields.